File: python/collatz-conjecture/collatz_conjecture.py
def steps(number):
    if number < 1:
        raise ValueError("The Collatz Conjecture is only concerned with strictly positive integers.")
    
    steps = 0
    while number!= 1:
        if number % 2 == 0:
            number /= 2
        else:
            number = (3*number) + 1
        steps += 1
    return steps

File: r/collatz-conjecture/collatz-conjecture.R
collatz_step_counter <- Vectorize(function(num) {
  if(num < 1) {
    stop("The Collatz sequence only terminates for positive numbers.")
  }

  if (num == 1) {
    0
  } else {
    1 + collatz_step_counter(ifelse(num %% 2 == 0, num / 2, 3*num + 1))
  }
})
